### PR TITLE
Add support for Composer in purl2url

### PR DESCRIPTION
**Pull request by camillem**
Hi,

I have tried to add support for Composer to purl2url. 
It might be relevant to https://github.com/package-url/packageurl-python/issues/143 

Cheers, 
Camille

### CODE REVIEW

**Comment by tdruez** [comment]
@camillem That's great! Could you add the url2purl side as well?

**Comment by camillem** [comment]
> @camillem That's great! Could you add the url2purl side as well?

Yep, I'll do that.

File: src/packageurl/contrib/purl2url.py [hunk 1]
```diff
@@ -225,6 +225,23 @@ def build_pypi_repo_url(purl):
...
 
 
+@repo_router.route("pkg:composer/.*")
+def build_composer_repo_url(purl):
+    """
+    Return a composer repo URL from the `purl` string.
+    """
+    purl_data = PackageURL.from_string(purl)
+
+    name = (purl_data.name or "").replace("_", "-")
```
Comment (tdruez) [inline src/packageurl/contrib/purl2url.py:235]: I'm that sure that an empty name is ever possible, see https://github.com/camillem/packageurl-python/blob/e0d180d1ce23f289230a970c2d1e560456300bb8/src/packageurl/__init__.py#L542

Comment (camillem) [reply src/packageurl/contrib/purl2url.py:235]: Indeed. I just directly copy/pasted it from https://github.com/camillem/packageurl-python/blob/e0d180d1ce23f289230a970c2d1e560456300bb8/src/packageurl/contrib/purl2url.py#L219
I should also check if the `replace("_", "-")` makes sense in this context.

Comment (camillem) [reply src/packageurl/contrib/purl2url.py]: I simplified this part, as the `replace("_", "-")`  seemed to be linked to an old pypi.org oddity.

**Comment by tdruez** [comment]
Thanks @camillem for your contribution 🚀

**Pull request merged**